scripts/convert_pretrained_checkpoint.py

```diff
@@ -0,0 +1,83 @@
+import json
+import shutil
+import sys
+from dataclasses import asdict
+from pathlib import Path
+
+import torch
+
+# support running without installing as a package
+wd = Path(__file__).parent.parent.resolve()
+sys.path.append(str(wd))
+
+from lit_gpt import Config
+from lit_gpt.utils import incremental_save, lazy_load
+
+
+@torch.inference_mode()
+def convert_checkpoint(
+    checkpoint_file: Path,
+    tokenizer_path: Path,
+    config_name: str,
+    output_folder: Path,
+    model_key: str = "model",
```

Comment:
Many of our scripts hardcode this model key already:

https://github.com/search?q=repo%3ALightning-AI%2Flit-gpt%20.get(%22model&type=code

If this script is added, then those uses should be removed and require this conversion step (not only pretrained).

But then there's the concern of disk usage. A naive user could have

1. The original HF weights
2. The converted HF weights
3. The pretrained/finetuned checkpoint(s)
4. The converted pretrained/finetuned checkpoint(s)

This makes me wonder if we should instead have a function that does this conversion on demand and is used in all scripts that would need it.

What do you think? I advocate for a holistic solution to this problem that works across scripts and trainer/fabric.